stdlib/ballerina-http/src/main/java/org/ballerinalang/net/uri/URIUtil.java

```diff
@@ -39,4 +45,21 @@ public static String getSubPath(String path, String basePath) {
 
         return path.substring(basePath.length());
     }
+
+    public static void populateQueryParamMap(String payload, BMap<String, BString> params)
+            throws UnsupportedEncodingException {
+        String[] entries = payload.split("&");
+        for (String entry : entries) {
```

Comment:
Here for the entry how about using quaryPrams ? IMO, that would increase the readability.